usr/src/uts/common/os/sysent.c

```diff
@@ -434,329 +429,277 @@ struct sysent nosys_ent = SYSENT_NOSYS();
-	/* 20 */ SYSENT_2CI("getpid",		getpid,		0),
-	/* 21 */ SYSENT_AP("mount",		mount,		8),
-	/* 22 */ SYSENT_CL("readlinkat",	readlinkat,	4),
-	/* 23 */ SYSENT_CI("setuid",		setuid,		1),
-	/* 24 */ SYSENT_2CI("getuid",		getuid,		0),
-	/* 25 */ SYSENT_CI("stime",		stime,		1),
-	/* 26 */ SYSENT_CL("pcsample",		pcsample,	2),
-	/* 27 */ SYSENT_CI("alarm",		alarm,		1),
-	/* 28 */ SYSENT_CI("fstat",		fstat,		2),
-	/* 29 */ SYSENT_CI("pause",		pause,		0),
-	/* 30 */ SYSENT_LOADABLE(),			/* (was utime) */
-	/* 31 */ SYSENT_CI("stty",		stty,		2),
-	/* 32 */ SYSENT_CI("gtty",		gtty,		2),
-	/* 33 */ SYSENT_CI("access",		access,		2),
-	/* 34 */ SYSENT_CI("nice",		nice,		1),
-	/* 35 */ IF_LP64(
-			SYSENT_NOSYS(),
-			SYSENT_CI("statfs",	statfs32,	4)),
-	/* 36 */ SYSENT_CI("sync",		syssync,	0),
-	/* 37 */ SYSENT_CI("kill",		kill,		2),
-	/* 38 */ IF_LP64(
-			SYSENT_NOSYS(),
-			SYSENT_CI("fstatfs",	fstatfs32,	4)),
-	/* 39 */ SYSENT_CI("setpgrp",		setpgrp,	3),
-	/* 40 */ SYSENT_CI("uucopystr",		uucopystr,	3),
-	/* 41 */ SYSENT_LOADABLE(),			/* (was dup) */
-	/* 42 */ SYSENT_LOADABLE(),			/* pipe */
-	/* 43 */ SYSENT_CL("times",		times,		1),
-	/* 44 */ SYSENT_CI("prof",		profil,		4),
-	/* 45 */ SYSENT_CI("faccessat",		faccessat,	4),
-	/* 46 */ SYSENT_CI("setgid",		setgid,		1),
-	/* 47 */ SYSENT_2CI("getgid",		getgid,		0),
-	/* 48 */ SYSENT_CI("mknodat",		mknodat,	4),
-	/* 49 */ SYSENT_LOADABLE(),			/* msgsys */
-	/* 50 */ IF_x86(
+	[0]	= SYSENT_NOSYS(),	/* Was indirect syscall on 32bit */
+	[1]	= SYSENT_CI("exit",		rexit,		1),
+	[2]	= SYSENT_CI("psecflags",	psecflags,	3),
+	[3]	= SYSENT_CL("read",		read,		3),
+	[4]	= SYSENT_CL("write",		write,		3),
+	[5]	= SYSENT_CI("open",		open,		3),
+	[6]	= SYSENT_CI("close",		close,		1),
+	[7]	= SYSENT_CI("linkat",		linkat,		5),
+	[8]	= SYSENT_LOADABLE(),		/* (was creat) */
+	[9]	= SYSENT_CI("link",		link,		2),
+	[10]	= SYSENT_CI("unlink",		unlink,		1),
+	[11]	= SYSENT_CI("symlinkat",	symlinkat,	3),
+	[12]	= SYSENT_CI("chdir",		chdir,		1),
+	[13]	= SYSENT_CL("time",		gtime,		0),
+	[14]	= SYSENT_CI("mknod",		mknod,		3),
+	[15]	= SYSENT_CI("chmod",		chmod,		2),
+	[16]	= SYSENT_CI("chown",		chown,		3),
+	[17]	= SYSENT_CI("brk",		brk,		1),
+	[18]	= SYSENT_CI("stat",		stat,		2),
+	[19]	= SYSENT_CL("lseek",		lseek64,	3),
+	[20]	= SYSENT_2CI("getpid",		getpid,		0),
+	[21]	= SYSENT_AP("mount",		mount,		8),
+	[22]	= SYSENT_CL("readlinkat",	readlinkat,	4),
+	[23]	= SYSENT_CI("setuid",		setuid,		1),
+	[24]	= SYSENT_2CI("getuid",		getuid,		0),
+	[25]	= SYSENT_CI("stime",		stime,		1),
+	[26]	= SYSENT_CL("pcsample",		pcsample,	2),
+	[27]	= SYSENT_CI("alarm",		alarm,		1),
+	[28]	= SYSENT_CI("fstat",		fstat,		2),
+	[29]	= SYSENT_CI("pause",		pause,		0),
+	[30]	= SYSENT_LOADABLE(),		/* (was utime) */
+	[31]	= SYSENT_CI("stty",		stty,		2),
+	[32]	= SYSENT_CI("gtty",		gtty,		2),
+	[33]	= SYSENT_CI("access",		access,		2),
+	[34]	= SYSENT_CI("nice",		nice,		1),
+	[35]	= SYSENT_NOSYS(),		/* (was 32bit statfs) */
```

Comment:
Given that we no longer have the 32-bit version is there anything that stops these from becoming loadable at some point in the future? I'm not asking for that here and I suspect some time64 stuff may eat it up, but wanted to check.

Reply:
These system call numbers are still used for 32bit processes, and it seemed unwise to allow the syscall tables to potentially diverge in the meaning of a given number.

Saying "We don't have `stat32()`" seemed fine, but saying "syscall 35 is `stat32` on ia32 but `something_new` on amd64" seemed like a disaster.  It's also impossible to state in `/etc/name_to_sysnum` (which is its own problem).

Reply:
That makes sense, thanks. I hadn't consulted the other table and I agree that consistency is important.